src/reactor/mod.rs

```diff
@@ -747,7 +748,7 @@ impl<F: FnOnce(&Core) + Send + 'static> FnBox for F {
 }
 
 fn read_ready() -> mio::Ready {
-    mio::Ready::readable() | platform::hup()
+    mio::Ready::readable() | platform::hup() | platform::aio()
```

Comment:
I think this change may not be necessary? The thinking here for `hup` was that pipes on Unix and such typically use `hup` as "this is now readable at eof" and they may not otherwise generate a readable event (I think there's a test case for this?) but for general I/O objects I don't think `aio` should by in the default read events?

You'd still get notifications, though, through `poll_ready` of course.